cubeviz/layout.py

```diff
@@ -313,6 +330,25 @@ def _enable_all_viewer_combos(self, data):
             view = self.all_views[i].widget()
             view.update_axes_title(str(getattr(self, selection_label)))
 
+    def change_viewer_component(self, view_index,
+                                component_index,
+                                force=False):
+        """
+        Given a viewer at an index view_index, change combo
+        selection to component at an index component_index.
+        :param view_index: int: Viewer index
+        :param component_index: int: Component index in viewer combo
+        :param force: bool: force change if component is already displayed.
+        """
+        if view_index == 0:
+            combo_label = 'single_viewer_combo'
+        else:
+            combo_label = 'viewer{0}_combo'.format(view_index)
+        combo = getattr(self.ui, combo_label)
+        if combo.currentIndex() == component_index and force:
+            combo.currentIndexChanged.emit(component_index)
+        else:
+            combo.setCurrentIndex(component_index)
```

Comment:
Is it possible that both cases of this branch can just use `combo.currentIndexChanged.emit`?

Reply:
We need `combo.setCurrentIndex(component_index)` to change the index of the combo box. The first one is to activate all the functions listening.